### PR TITLE
Dynamic Simulation - Model and Parameter Definitions in many-to-many with extra column 'origin'

### PR DESCRIPTION
**Pull request by thangqp**
'Origin' of a Parameter definition may be different in relation to different models, for example, 'USER' to 'NETWORK'

So 'Origin' will be persisted in the relation table with an extra-column 'origin' 

In this PR: 

- rename column 'model_name' to 'name' in table 'models', for being compatible to the Hibernate generated sqls
- provide a relation entity for ModelEntity and ModelParameterDefinitionEntity which contain field 'origin'. This entity represents many-to-many relation with extra-column
- Add Delete Endpoint for Model. This is a delete in cascade for the whole model.

### CODE REVIEW

File: src/main/java/org/gridsuite/mapping/server/model/ModelEntity.java [hunk 1]
```diff
@@ -85,14 +83,28 @@ public ModelEntity(Model modelToConvert) {
...
+
+    public void addParameterDefinitions(Collection<ModelParameterDefinitionEntity> parameterDefinitions, ParameterOrigin origin) {
+        parameterDefinitions.forEach(parameterDefinition -> addParameterDefinition(parameterDefinition, origin));
+    }
+
+    public void removeParameterDefinition(ModelParameterDefinitionEntity parameterDefinition) {
+        for (Iterator<ModelModelParameterDefinitionEntity> iter = this.parameterDefinitions.iterator(); iter.hasNext();) {
+            ModelModelParameterDefinitionEntity modelModelParameterDefinitionEntity = iter.next();
+            if (modelModelParameterDefinitionEntity.getModel().equals(this) && modelModelParameterDefinitionEntity.getParameterDefinition().equals(parameterDefinition)) {
+                iter.remove();
```
Comment (jonenst) [inline src/main/java/org/gridsuite/mapping/server/model/ModelEntity.java:99]: do you need to remove all occurences or can you break on first find ?

Comment (jonenst) [reply src/main/java/org/gridsuite/mapping/server/model/ModelEntity.java:99]: either add break, or rename the function removeAllParameterDefinition and add comments ?

File: src/main/java/org/gridsuite/mapping/server/model/ModelParameterDefinitionEntity.java [hunk 2]
```diff
@@ -43,25 +39,14 @@ public class ModelParameterDefinitionEntity implements Serializable {
...
-    @Enumerated
-    private ParameterOrigin origin;
-
     @Column(name = "origin_name")
     private String originName;
 
     @Column(name = "fixed_value")
     private String fixedValue;
 
-    @ManyToMany(
```
Comment (jonenst) [inline src/main/java/org/gridsuite/mapping/server/model/ModelParameterDefinitionEntity.java:56]: this was not needed ? or do you need ontomany to modelmodelparameterdefinitionentity ?

File: src/main/resources/db/changelog/changesets/changelog_20230908T150155Z.xml [hunk 3]
```diff
@@ -0,0 +1,99 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-4.1.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.1.xsd">
+    <changeSet author="phamquy (generated)" id="1694185324695-0">
+        <preConditions onFail="MARK_RAN">
+            <columnExists tableName="models" columnName="model_name"/>
+        </preConditions>
+        <comment>Clear any existing legacy data in model-related tables to allow the next step which drops the column model_name in primary key of table models</comment>
+        <sql splitStatements="true">
+            DELETE FROM model_parameters;
```
Comment (jonenst) [inline src/main/resources/db/changelog/changesets/changelog_20230908T150155Z.xml:9]: implement the same thing as last time, don't remember what it was but I don't think we did mass deletes in liquibase like this ?

**Comment by jonenst** [comment]
> rename column 'model_name' to 'name' in table 'models', for being compatible to the Hibernate generated sqls

Can you explain what is not compatible ?

File: src/main/java/org/gridsuite/mapping/server/model/ModelParameterDefinitionEntity.java (hunk 2 above)
Comment (thangqp) [reply src/main/java/org/gridsuite/mapping/server/model/ModelParameterDefinitionEntity.java:56]: Bi-directional relation is not really needed as there is any business logic which requires to navigate from parameter definition to model, that why in this technical PR I have removed this reverse direction..

Comment (jonenst) [reply src/main/java/org/gridsuite/mapping/server/model/ModelParameterDefinitionEntity.java:56]: ok

File: src/main/java/org/gridsuite/mapping/server/model/ModelEntity.java (hunk 1 above)
Comment (thangqp) [reply src/main/java/org/gridsuite/mapping/server/model/ModelEntity.java:99]: You are right! we can use first found, I will correct it and rename for add all function..

Comment (thangqp) [reply src/main/java/org/gridsuite/mapping/server/model/ModelEntity.java:113]: Done

File: src/main/java/org/gridsuite/mapping/server/model/ModelParameterDefinitionEntity.java (hunk 2 above)
Comment (thangqp) [reply src/main/java/org/gridsuite/mapping/server/model/ModelParameterDefinitionEntity.java:56]: I have put OneToMany relation back for facilitating to delete a model in cascade manually, see the new endpoint

File: src/main/resources/db/changelog/changesets/changelog_20230908T150155Z.xml (hunk 3 above)
Comment (thangqp) [reply src/main/resources/db/changelog/changesets/changelog_20230908T150155Z.xml]: This solution is cancelled...

**Review by jonenst — APPROVED** [review]

**Pull request merged**